frontend/hooks/data-fetching/dashboard.ts

```diff
@@ -19,6 +19,22 @@ export async function joinCourse(courseId: string): Promise<void> {
     }
 }
 
+export async function leaveCourse(
+    courseId: string,
+    membershipId: string
+): Promise<void> {
+    const res = await doApiRequest(
+        `/courses/${courseId}/members/${membershipId}`,
```

Comment:
I haven't looked at the rest of this, but all API requests should have a trailing slash